src/main/java/org/openrewrite/java/spring/NoAutowiredOnConstructor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.RemoveAnnotation;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+
+/**
+ * Removes the Autowired annotation from a bean's constructor when there is only one constructor.
+ */
+public class NoAutowiredOnConstructor extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Remove the `@Autowired` annotation on inferred constructor";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Spring can infer an autowired constructor when there is a single constructor on the bean.";
+    }
+
+    @Nullable
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesType<>("org.springframework.beans.factory.annotation.Autowired");
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new NoAutowiredOnConstructor.NoAutowiredOnConstructorVisitor();
+    }
+
+    private static class NoAutowiredOnConstructorVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext executionContext) {
+            int constructorCount = 0;
+
+            for (Statement s : classDecl.getBody().getStatements()) {
+                if (s instanceof J.MethodDeclaration) {
+                    if (((J.MethodDeclaration) s).isConstructor()) {
+                        constructorCount++;
+                    }
+                }
+            }
+
+            if (constructorCount == 1) {
+                doAfterVisit(new RemoveAnnotation("@org.springframework.beans.factory.annotation.Autowired(true)"));
```

Comment:
This will remove ALL auto-wired annotations from the file if one constructor exists. We ONLY want to remove the auto-wired from the Constructor and leave all other annotations alone.